crates/bevy_gltf/src/loader/mod.rs

```diff
@@ -1899,6 +1935,211 @@ struct MorphTargetNames {
+            buffer_data,
+            node_index_to_parent_node_index,
+            node_index_to_global_transform: HashMap::default(),
+        }
+    }
+
+    /// Computes all global transforms needed to compute AABBs.
+    ///
+    /// Since computing a global transform can be mildly expensive, we avoid
+    /// doing it for nodes other than those that will be needed to properly
+    /// compute bounding boxes. We must compute the transform for each node that
+    /// contains a skinned mesh and each node that represents a joint.
+    fn compute_needed_global_transforms(&mut self, gltf: &gltf::Gltf) {
+        // We don't need to compute any global transforms if there are no skins
+        // in the glTF file.
+        if gltf.skins().next().is_none() {
+            return;
+        }
+
+        for node in gltf.nodes() {
+            if let Some(skin) = node.skin() {
+                self.compute_global_transform_for_node(gltf, node.index());
+                for joint in skin.joints() {
+                    self.compute_global_transform_for_node(gltf, joint.index());
+                }
+            }
+        }
+    }
+
+    /// A helper method that computex the global transform for a node and its
+    /// ancestors if necessary.
+    fn compute_global_transform_for_node(&mut self, gltf: &gltf::Gltf, node_index: usize) {
+        // If we've already computed the global transform for this node, bail.
+        if self
+            .node_index_to_global_transform
+            .contains_key(&node_index)
+        {
+            return;
+        }
+
+        // Compute this node's local transform. We can get away with `Affine3A`s
+        // because the glTF spec requires that matrices "*MUST* be decomposable
+        // to TRS properties".
+        let local_transform = Affine3A::from_mat4(Mat4::from_cols_array_2d(
+            &gltf.nodes().nth(node_index).unwrap().transform().matrix(),
+        ));
+
+        // If we have a parent, then compute its global transform and append
+        // ours. Otherwise, if we're at top level, the global transform is just
+        // the node's transform.
+        match self.node_index_to_parent_node_index.get(&node_index) {
+            Some(&parent_index) => {
+                self.compute_global_transform_for_node(gltf, parent_index);
+                self.node_index_to_global_transform.insert(
+                    node_index,
+                    self.node_index_to_global_transform[&parent_index] * local_transform,
+                );
+            }
+            None => {
+                self.node_index_to_global_transform
+                    .insert(node_index, local_transform);
+            }
+        }
+    }
+
+    /// Computes and returns the joint matrices for the given skinned mesh if
+    /// applicable.
+    ///
+    /// If the given node doesn't represent a skinned mesh, returns an empty
+    /// vector.
+    fn compute_joint_matrices_for_node(&mut self, gltf_node: &Node) -> Vec<Affine3A> {
```

Comment:
```suggestion
    /// If the given node doesn't represent a skinned mesh, returns `None`.
    fn compute_joint_matrices_for_node(&mut self, gltf_node: &Node) -> Option<Vec<Affine3A>> {
```

Changing to an `Option` would add complication, but maybe worth it to clarify the non-skinned case?